configuration_files/trajectory_builder.lua

```diff
@@ -19,4 +19,7 @@ TRAJECTORY_BUILDER = {
   trajectory_builder_2d = TRAJECTORY_BUILDER_2D,
   trajectory_builder_3d = TRAJECTORY_BUILDER_3D,
```

Comment:
can we get rid of `pure_localization` flag in lua and proto as well? having `pure_localization_trimmer` should be enough

Reply:
So check if there is a `pure_localization_trimmer` and depending on that go into `pure_localization` mode?
Do you want that in this PR?

Reply:
yes

Reply:
Finally got around to do the changes.

Please check.